test/src/tests-batchUploader.ts

```diff
@@ -1716,5 +1717,113 @@ describe('batch uploader', () => {
         .catch((e) => {
         })
         });
+
+        describe('noFunctional', () => {
+            const eventStorageKey = 'mprtcl-v4_abcdef-events';
+            const batchStorageKey = 'mprtcl-v4_abcdef-batches';
+
+            it('should write session storage when noFunctional is default (false)', (done) => {
+                window.mParticle._resetForTests(MPConfig);
+                window.mParticle.config.flags = {
+                    offlineStorage: '100',
+                    ...enableBatchingConfigFlags,
+                };
+                window.mParticle.init(apiKey, window.mParticle.config);
+                waitForCondition(hasIdentifyReturned)
+                .then(() => {
+                    const mpInstance = window.mParticle.getInstance();
+                    const uploader = mpInstance._APIClient.uploader;
+                    uploader.queueEvent(event0);
+                    expect(!!window.sessionStorage.getItem(eventStorageKey)).to.equal(true);
+                    done();
+                })
+                .catch(done);
+            });
```

Comment:
Try to use `async/await` instead of `then` for these new tests.

```suggestion
            it('should write session storage when noFunctional is default (false)', async () => {
                window.mParticle._resetForTests(MPConfig);
                window.mParticle.config.flags = {
                    offlineStorage: '100',
                    ...enableBatchingConfigFlags,
                };
                window.mParticle.init(apiKey, window.mParticle.config);
                await waitForCondition(hasIdentifyReturned);

                const mpInstance = window.mParticle.getInstance();
                const uploader = mpInstance._APIClient.uploader;
                uploader.queueEvent(event0);
                expect(!!window.sessionStorage.getItem(eventStorageKey)).to.equal(true);
            });
```